### PR TITLE
feat: upgrades geo category to cdkv2

### PR DESCRIPTION
**Pull request by akshbhu**
#### Description of changes
-upgrades geo category to V2
- lint fixes

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-geo/package.json [hunk 1]
```diff
@@ -22,15 +22,14 @@
     "aws"
   ],
   "dependencies": {
-    "@aws-cdk/aws-iam": "^1.159.0",
-    "@aws-cdk/aws-lambda": "^1.159.0",
-    "@aws-cdk/core": "^1.159.0",
+    "aws-cdk-lib": "^2.41.0",
```
Comment (sobolk) [inline packages/amplify-category-geo/package.json:25]: let's bump major version of geo category as well.

File: packages/amplify-category-geo/src/__tests__/service-stacks/__snapshots__/mapStack.test.ts.snap [hunk 2]
```diff
@@ -112,10 +112,6 @@ Object {
       "UpdateReplacePolicy": "Delete",
     },
     "CustomMapLambda51D5D430": Object {
-      "DependsOn": Array [
```
Comment (sobolk) [inline packages/amplify-category-geo/src/__tests__/service-stacks/__snapshots__/mapStack.test.ts.snap:115]: what's the purpose of this file `mapStack.test.ts.snap`?

File: packages/amplify-category-geo/src/service-stacks/mapStack.ts [hunk 3]
```diff
@@ -100,7 +108,7 @@ export class MapStack extends BaseStack {
       resourceType: 'Custom::LambdaCallout',
       properties: {
         mapName: this.mapName,
-        mapStyle: mapStyle,
+        mapStyle,
```
Comment (sobolk) [inline packages/amplify-category-geo/src/service-stacks/mapStack.ts:111]: is this intentional ?

**Comment by akshbhu** [comment]
Closing in favour of https://github.com/aws-amplify/amplify-cli/pull/10988

**Pull request closed without merging**